examples/s/d/nn/_project/aggregate_surfaces.py

```diff
@@ -0,0 +1,236 @@
+           context, the service itself is responsible for storing the results.
+
+    Note that this example is showing the usage of fmu-dataio, it is not to be seen as
+    an example for the actual aggregation. The aggregation service shown here is
+    simplistic and its sole purpose is to facilitate the fmu-dataio example.
+
+    """
+
+    # First we get the input data (the individual surfaces from each realization), which
+    # we assume are stored in classical FMU style on /scratch disk folder structure.
+
+    # IRL, these variables would typically be arguments to the aggregation script.
+    casepath = Path("../xcase/").resolve()
+    iter_name = "iter-0"
+    relative_path = (
+        "share/results/maps/topvolantis--ds_extract_geogrid.gri"  # exists in all reals
+    )
+    realization_ids = _get_realization_ids(casepath)
+
+    # gather source surfaces and their associated metadata
+    source_surfaces, source_metadata = _get_source_surfaces_from_disk(
+        casepath=casepath,
+        iter_name=iter_name,
+        realization_ids=realization_ids,
+        relative_path=relative_path,
+    )
+
+    # These are the operations we want to do
+    operations = ["mean", "min", "max", "std"]
+
+    # This is the ID we assign to this set of aggregations
+    aggregation_id = "something_very_unique"  # IRL this will usually be a uuid
+
+    # We aggregate these source surfaces and collect results in list of dictionaries
+    aggregations = []
+
+    # Initialize an AggregatedData object for this set of aggregations
+    exp = fmu.dataio.AggregatedData(
+        source_metadata=source_metadata,
+        aggregation_id=aggregation_id,
+        casepath=casepath,
+    )
+
+    for operation in operations:
+
+        # Call the aggregation machine and create an aggregated surface
+        # Note that this is not part of fmu-dataio - it is merely a mock-up
+        # aggregation service for the sake of this example.
+        aggregated_surface = _aggregate(source_surfaces, operation)
+
+        # ==============================================================================
+        # Example 1: We want fmu-dataio to export the file + metadata to disk
+        saved_filename = exp.export(aggregated_surface, operation=operation)
+        print(f"file saved to {saved_filename}")
+
+        # ==============================================================================
+        # Example 2: We only want the metadata (e.g. we are in a cloud service)
+        metadata = exp.generate_metadata(aggregated_surface, operation=operation)
+
+        # At this point, we have the surface, the operation and the metadata
+        # These can be collected into e.g. a list or a dictionary for further usage,
+        # or we can upload to Sumo (not implemented in this example).
+
+        # For now, we collect them in the 'aggregations' list for sake of the example.
+        aggregations.append(
+            {
+                "surface": aggregated_surface,
```

Comment:
A bit unclear what the point of this is?

Reply:
It is a pattern used in some existing implementations, but I agree that in the example it becomes a bit redundant. Have removed it.